spec/factories/discussions.rb

```diff
@@ -3,5 +3,34 @@
 FactoryBot.define do
   factory :discussion do
     hidden { false }
+
+    trait :with_0to4_comments do
+      after :create do |discussion|
+        create_list :comment, rand(5), discussion: discussion
+      end
+    end
+
+    trait :with_four_comments do
```

Comment:
Some of these factories look like they could be consolidated using factorybot's transient attributes. Consider introducing a transient attribute for `comments_count`.